HTML/HTML.sublime-syntax

```diff
@@ -19,20 +19,88 @@ first_line_match: (?i)<(!DOCTYPE\s*)?html
 ###############################################################################
 
 variables:
-  block_tag_name: |-
-    (?ix:
-      address|applet|article|aside|blockquote|center|dd|dir|div|dl|dt|figcaption|figure|footer|frame|frameset|h1|h2|h3|h4|h5|h6|header|iframe|menu|nav|noframes|object|ol|p|pre|section|ul
-    ){{tag_name_break}}
+  # HTML tags
+  # https://developer.mozilla.org/en-US/docs/Web/HTML/Element
+  # Note: Sections are sorted by expected frequency.
 
-  inline_tag_name: |-
-    (?ix:
-      abbr|acronym|area|audio|b|base|basefont|bdi|bdo|big|br|canvas|caption|cite|code|del|details|dfn|dialog|em|font|head|html|i|img|ins|isindex|kbd|li|link|map|mark|menu|menuitem|meta|noscript|param|picture|q|rp|rt|rtc|ruby|s|samp|script|small|source|span|strike|strong|style|sub|summary|sup|time|title|track|tt|u|var|video|wbr
-    ){{tag_name_break}}
+  html_block_tags: |-
+    (?x:
+      {{html_text_content_tags}}
+    | {{html_section_tags}}
+    | {{html_embedded_tags}}
+    | {{html_web_tags}}
+    | {{html_header_tags}}
+    | {{html_deprecated_block_tags}}
+    )
+
+  html_inline_tags: |-
+    (?x:
+      {{html_text_semantic_tags}}
+    | {{html_markup_tags}}
+    | {{html_media_tags}}
+    | {{html_interactive_tags}}
+    | {{html_script_tags}}
+    | {{html_deprecated_inline_tags}}
+    )
+
+  # Aggregates structural root notes from
+  # https://developer.mozilla.org/en-US/docs/Web/HTML/Element#main_root
+  # https://developer.mozilla.org/en-US/docs/Web/HTML/Element#document_metadata
+  # https://developer.mozilla.org/en-US/docs/Web/HTML/Element#sectioning_root
+  html_structure_tags: |-
+    (?xi: html | head | body ){{tag_name_break}}
+
+  # Note: <head> is moved to `html_structure_tags`
+  html_header_tags: |-
+    (?xi: base | link | meta | (?# script | style | ) title ){{tag_name_break}}
+
+  html_section_tags: |-
+    (?xi: address | article | aside | footer | header | h[1-6] | hgroup | main
+    | nav | section ){{tag_name_break}}
+
+  html_text_content_tags: |-
+    (?xi: blockquote | cite | dd | dt | dl | div | figcaption | figure | hr
+    | li | ol | p | pre | ul ){{tag_name_break}}
+
+  html_text_semantic_tags: |-
+    (?xi: a | abbr | b | bdi | bdo | br | code | data | time | dfn | em | i
+    | kbd | mark | q | ruby | rp | rt | s | samp | small | span | strong | sub
+    | sup | u | var | wbr ){{tag_name_break}}
+
+  html_media_tags: |-
+    (?xi: area | audio | img | map | track | video ){{tag_name_break}}
+
+  html_embedded_tags: |-
+    (?xi: embed | iframe | object | param | picture | source ){{tag_name_break}}
+
+  html_script_tags: |-
+    (?xi: canvas | noscript (?# | script ) ){{tag_name_break}}
```

Comment:
it seems odd to see `script` tag commented out in a context called `html_script_tags`. Can you explain the reasoning, please?
Follow up question, why comment them out instead of just delete them if they are not needed?

Reply:
All variables mainly exist to keep track of MDN changes more easily. The `script` tag is handled by a dedicated context `script-tag` and is listed twice on MDN - in header tags and script tags section. By commenting out, I wanted to make clear it was not forgotten/missed but disabled by intent.